### ISSUE TITLE
ICE in CompilerStack: contract redefined to inherit from a non-existent type not detected as name clash

### ISSUE THREAD

**Issue by agroce** [ethereum/solidity#9948]
## Description

This contract:
```solidity
contract X {}
library test {
}
contract X is A {
}
```

produces:

```
Internal compiler error during compilation:
/home/user/solidity/libsolidity/interface/CompilerStack.cpp(1131): Throw in function void solidity::frontend::CompilerStack::storeContractDefinitions()
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: Contract already present (name clash?), but no error was reported.
[solidity::util::tag_comment*] = Contract already present (name clash?), but no error was reported.
```

when compiled with solc.

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer.

## Environment

- Compiler version:  0.7.3-develop.2020.10.2+commit.756e21a8.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

**Comment by cameel** [comment]
Thanks for the report! Looks like it's something that broke very recently. There's no ICE on 0.7.2 (see output below).

I see that #9364 was merged after 0.7.2 had been relased and it did touch the assert that fails here so I suspect it has something to do with it (pinging @Marenz and @chriseth and moving to implementation backlog).

### Output on 0.7.2
```solidity
Error: Identifier already declared.
 --> /tmp/test.sol:4:1:
  |
4 | contract X is A {
  | ^ (Relevant source part starts here and spans across multiple lines).
Note: The previous declaration is here:
 --> /tmp/test.sol:1:1:
  |
1 | contract X {}
  | ^^^^^^^^^^^^^

Error: Identifier not found or not unique.
 --> /tmp/test.sol:4:15:
  |
4 | contract X is A {
  |  
```

**Comment by agroce** [comment]
Yeah, it's just taking broken code from a normal failure to compile to an ICE, but I figured worth fixing up, in case it had any deeper meaning, as well (since it is allowing this code to get past the stage where compile aborts on errors, which seems a little concerning).

**Comment by agroce** [comment]
Also, fixing it will help fuzzing; it's so easy to trigger, AFL is spending a lot of time dumping out crash duplicates of this... :)

**Comment by agroce** [comment]
TY for the label :)

**Comment by cameel** [comment]
Sure, I marked it as `cosmetic` simply to indicate that the code is not valid and would result in a compiler error even without the ICE. Just one of the labels we created recently to sort bugs into categories. Doesn't mean it it's not important :) Your report is a great help because it's a bug that slipped through code review of a PR we just merged. Thanks to your report we'll be able to fix it even before it gets into an actual release.

> Also, fixing it will help fuzzing; it's so easy to trigger, AFL is spending a lot of time dumping out crash duplicates of this... :)

Yeah, stuff like this affects @bshastry's fuzzing too so I get what you mean. We even have a label to mark these bugs as high priority: `fuzz-blocker`. I think it would be fine for you to use it for stuff that affects your fuzzer too.
